utilities/cassandra/partition_meta_data.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include "rocksdb/db.h"
+#include "rocksdb/slice.h"
+#include "util/arena.h"
+#include "util/dynamic_bloom.h"
+#include "utilities/cassandra/format.h"
+namespace rocksdb {
+namespace cassandra {
+
+/**
+ * PartitionMetaData is for managing meta data (such as partition deletion) for
+ * each cassandra partitions. It should be initialized per rocksdb instance.
+ */
+class PartitionMetaData {
+ public:
+  PartitionMetaData(DB* db, ColumnFamilyHandle* meta_cf_handle,
+                    size_t token_length);
+
+  // Enable meta key bloom filter to filter out none exists key
+  // quickly. This helps boost performance when number of deleted
+  // partition key is only fraction of number of all keys
+  Status EnableBloomFilter(uint32_t bloom_total_bits_);
```

Comment:
Did you know that RocksDB supports memtable bloom filters? 
https://github.com/facebook/rocksdb/blob/4fce44fc8b8184f4800a792cf2753667e9efc9b3/include/rocksdb/advanced_options.h#L274-L282

I wonder if you can get the same performance for your meta column family when you enable memtable bloom filters + regular bloom filters. Did you try that?

Reply:
And if I assume that your metadata column family is much much smaller than the regular/other column family, you could set cache_index_and_filter_blocks=true and enable pinning for those blocks in this cf ... so all the bloom filter lookups could be satisfied from memory.

Reply:
the partition meta data will not be write very often so mostly on disk, so I wonder memtable filter will really help it or not. Will try it to see any difference anyway

Reply:
What about read? I believe it would be read frequently, isn't it? The filters will optimize for read performance, especially for negative queries (where the queried keys don't exist).